dexlib2/src/main/java/com/android/tools/smali/dexlib2/dexbacked/DexBackedOdexFile.java

```diff
@@ -85,14 +84,16 @@ public DexBackedOdexFile(@Nonnull Opcodes opcodes, @Nonnull byte[] odexBuf, byte
         DexUtil.verifyOdexHeader(is);
 
         is.reset();
-        byte[] odexBuf = new byte[OdexHeaderItem.ITEM_SIZE];
-        ByteStreams.readFully(is, odexBuf);
+        byte[] odexBuf = is.readNBytes(OdexHeaderItem.ITEM_SIZE);
         int dexOffset = OdexHeaderItem.getDexOffset(odexBuf);
         if (dexOffset > OdexHeaderItem.ITEM_SIZE) {
-            ByteStreams.skipFully(is, dexOffset - OdexHeaderItem.ITEM_SIZE);
+            is.skip(dexOffset - OdexHeaderItem.ITEM_SIZE);
```

Comment:
Same issue for skip here. The java impl may only read a smaller number for bytes. Using readNBytes might be the simplest replacement.